### ISSUE TITLE
Support websockets

### ISSUE THREAD

**Issue by pengux** [gorilla/rpc#27]
Currently, the package is hard coupled with net/http and only support the HTTP transport. As websockets is now widely supported in all modern browser, it would be great to use it with this package.

**Comment by initzero** [comment]
Agreed. Would be great it jsonrpc2 was usable with websocket connections.

**Comment by tejasmanohar** [comment]
FWIW, we were also looking for this at work but ended up using GRPC instead.
